File: 백준/Silver/15903. 카드 합체 놀이/카드 합체 놀이.cc
#include <iostream>
#include <queue>
using namespace std;

int main()
{
    cin.tie(NULL);
    cout.tie(NULL);
    ios_base::sync_with_stdio(false);

    long long cards, unite, x, score{0};
    priority_queue<long long, vector<long long>, greater<long long>> deck;

    cin >> cards >> unite;
    while (cards--)
    {
        cin >> x;
        deck.push(x);
    }
    while (unite--)
    {
        long long first = deck.top();
        deck.pop();
        long long second = deck.top();
        deck.pop();
        long long sum = first + second;
        deck.push(sum);
        deck.push(sum);
    }
    while (!deck.empty())
    {
        score += deck.top();
        deck.pop();
    }
    cout << score;
}

File: 백준/Silver/15903. 카드 합체 놀이/카드 합체 놀이.py
def min_score(cards, changes):
    for _ in range(changes):
        cards.sort()
        sum = cards[0] + cards[1]
        cards[0] = sum
        cards[1] = sum

    score = 0

    for n in cards:
        score += n

    return score


n, changes = map(int, input().split())
cards = list(map(int, input().split()))

print(min_score(cards, changes))
